cpp/include/cuspatial/detail/geometry_collection/multilinestring_ref.cuh

```diff
@@ -74,13 +74,15 @@ CUSPATIAL_HOST_DEVICE auto multilinestring_ref<PartIterator, VecIterator>::part_
 template <typename PartIterator, typename VecIterator>
 CUSPATIAL_HOST_DEVICE auto multilinestring_ref<PartIterator, VecIterator>::point_begin() const
 {
-  return _point_begin;
+  return thrust::next(_point_begin, *_part_begin);
 }
 
 template <typename PartIterator, typename VecIterator>
 CUSPATIAL_HOST_DEVICE auto multilinestring_ref<PartIterator, VecIterator>::point_end() const
 {
-  return _point_end;
+  // _part_end is the one-past the last part index to the point of this multilinestring.
```

Comment:
```suggestion
  // _part_end refers to one past the last part index to the points of this multilinestring.
```
I don't quite understand this. A part index indexes points? Is it one past the last point of the part?

Reply:
See how multilinestring_ref is constructed: 

https://github.com/isVoid/cuspatial/blob/293e4c30b5b43ea57225fe4e52a55e18f4e99e3a/cpp/include/cuspatial/detail/range/multilinestring_range.cuh#L67

`*(_part_begin + _geometry_begin[i + 1])` is the part index of the last point of the current multilinestring. When constructing it, I increment it to maintain that the the end interator of a range should be one-past the last element.